__tests__/index.spec.js

```diff
@@ -214,11 +241,88 @@ describe('string-based items with output fn', () => {
     expect(rta.find('.rta__entity')).toHaveLength(0);
   });
 
-  it('should invoke onchange handler after selection', () => {
-    expect(mockedFn).toHaveBeenCalledTimes(2);
+  it('should invoke onChange handler after selection', () => {
+    expect(mockedChangeFn).toHaveBeenCalledTimes(2);
+  });
+
+  it('should invoke onCaretPositionChange handler after selection', () => {
+    expect(mockedCaretPositionChangeFn).toHaveBeenCalledTimes(2);
   });
 
   it('text in textarea should be changed', () => {
     expect(rta.find('textarea').node.value).toBe('__happy_face__some test :a');
   });
 });
+
+describe('using ref to the ReactTextareaAutocomplete to call methods', () => {
+  const mockedChangeFn = jest.fn();
+
+  class ReactTextareaAutocompleteWrapper extends Component {
```

Comment:
The wrapper is not necessary. You could use something like this. 

```js
const b = (
    <ReactTextareaAutocomplete
      placeholder={'Write a message.'}
      value={'Controlled text'}
      onChange={mockedChangeFn}
      className={'ownClassName'}
      style={{ background: 'red' }}
      loadingComponent={Loading}
      trigger={{
        ':': {
          dataProvider: () => Promise.resolve(['happy_face', 'sad_face']),
          component: SmileItemComponent,
        },
      }}
      ref={(ref) => {
        this.textareaRef = ref;
      }}
    />
  );

const rtaWrapper = mount(b);
```

Reply:
`this` will point to `global` here? I tried similar way before with a locally scoped variable and it wouldn't work correctly probably due to the lifecycle. Will give another chance ;)

Reply:
JSX will be transpiled into a function so `this` will point to this function. I tried to change the test like in the example above and everything passed ... maybe we have wrong tests 🤣